### PR TITLE
Feature/sc 467

### PR DESCRIPTION
**Pull request by CaptainOfHacks**

### CODE REVIEW

**Review by costezki — CHANGES_REQUESTED** [review]
the tests are failing, they need a fix.

**Review by Dragos0000 — APPROVED** [review]

File: sem_covid/adapters/rml_mapper.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Date:  13.12.2021
+# Author: Stratulat Ștefan
+# Email: stefan.stratulat1997@gmail.com
+
+import json
+from abc import ABC, abstractmethod
+import requests
+
+
+class RMLMapperABC(ABC):
```
Comment (costezki) [inline sem_covid/adapters/rml_mapper.py:13]: Need some documentation.

File: sem_covid/adapters/rml_mapper.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    @abstractmethod
+    def transform(self, rml_rule: str, sources: dict) -> str:
+        pass
+
+
+class RMLMapper(RMLMapperABC):
+
```
Comment (costezki) [inline sem_covid/adapters/rml_mapper.py:21]: document, purpose and usage.

File: sem_covid/entrypoints/etl_dags/rml_transform_dag.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+DEFAULT_DAG_ARGUMENTS = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime.now(),
+    "email": ["info@meaningfy.ws"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "retry_delay": timedelta(minutes=3600),
```
Comment (costezki) [inline sem_covid/entrypoints/etl_dags/rml_transform_dag.py:25]: 2.5 days? That is a bit too much :)

File: sem_covid/services/dataset_pipelines/rml_transform_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+MINIO_RML_RESULTS_DIR = 'results'
+DATASET_INDEX_NAME = 'ds_unified_dataset'
+DATASET_PART_SIZE = 100
+RDF_RESULT_FORMAT = 'nt11'
+
+
+
+class RMLTransformPipeline:
+    """
+
```
Comment (costezki) [inline sem_covid/services/dataset_pipelines/rml_transform_pipeline.py:24]: Documentation :)

File: sem_covid/services/dataset_pipelines/similarity_matrix_rml_transformation_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        self.rml_rule = self.object_storage.get_object(
+            object_name=f'{MINIO_RML_RULES_DIR}/{self.rml_rules_file_name}').decode('utf8')
+        dataset = self.feature_storage.get_features(features_name=self.source_file_name)
+        if self.use_sample_data:
+            self.dataset = dataset.head(100)
+        else:
+            self.dataset = dataset
+        df_size = len(self.dataset)
+        part_size = DATASET_PART_SIZE
+        number_of_parts = int(round(df_size / part_size, 0)) + 1
```
Comment (costezki) [inline sem_covid/services/dataset_pipelines/similarity_matrix_rml_transformation_pipeline.py:63]: Drop `round` and `int` and use `//` floor division operator instead.

Comment (costezki) [reply sem_covid/services/dataset_pipelines/similarity_matrix_rml_transformation_pipeline.py:63]: number_of_parts = len(self.dataset) // DATASET_PART_SIZE +1

File: sem_covid/services/dataset_pipelines/similarity_matrix_rml_transformation_pipeline.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            sources = {'ds_unified_sem_similarity_matrix.json': json_result}
+            self.rdf_results.append(self.rml_mapper.transform(rml_rule=self.rml_rule, sources=sources))
+        self.rdf_results = '\n'.join(self.rdf_results)  # this is the source of potential resource issues
+
+    def load(self):
+        """
+
+        :return:
+        """
+        assert self.rdf_results is not None
```
Comment (costezki) [inline sem_covid/services/dataset_pipelines/similarity_matrix_rml_transformation_pipeline.py:88]: :)

File: sem_covid/services/dataset_pipelines/similarity_matrix_rml_transformation_pipeline.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        self.triple_storage.create_dataset(dataset_id=DATASET_INDEX_NAME)
+        self.triple_storage.upload_triples(dataset_id=DATASET_INDEX_NAME, quoted_triples=self.rdf_results,
+                                           rdf_fmt=RDF_RESULT_FORMAT)
+
+    def execute(self):
+        """
+
+        :return:
+        """
+        self.extract()
```
Comment (costezki) [inline sem_covid/services/dataset_pipelines/similarity_matrix_rml_transformation_pipeline.py:100]: :) We really need to switch to a proper ETL architecture in Airflow :)

**Review by costezki — APPROVED** [review]

**Pull request merged**